herd/machModelChecker.ml

```diff
@@ -306,6 +306,22 @@ module Make
                  "F", E.is_barrier;
                  "DATA", is_data_port;
                  "NDATA", (fun e -> not (is_data_port e));])) in
+      let evts_map = E.EventSet.fold
+                (fun e evts_map ->
+                  let attrs = S.A.V.memattr_as_list (E.Act.memattr_of e.E.action) in
+                  List.fold_right
+                    (fun attr evts_map ->
+                      let evts_w_attr =
+                        match StringMap.find_opt attr evts_map with
+                        | Some evts -> E.EventSet.add e evts
+                        | None -> E.EventSet.add e E.EventSet.empty
+                      in
+                      let evts_map = StringMap.remove attr evts_map in
```

Comment:
This call is not useful as maps keep only one binding for a given key.

Reply:
Ack, I wasn't sure if the default behavior would be to keep or replace the old mapping. The code has changed a bit but the new version doesn't use `StringMap.remove`